From 256e32f94bf81b83251d936a6709fc56e0a91cd8 Mon Sep 17 00:00:00 2001
From: Tao Shaocheng <taoshaocheng@loongson.cn>
Date: Wed, 19 Apr 2023 10:38:43 +0800
Subject: [PATCH] [loong64][ptr-cage] Support shared-pointer-cage for loong64
 in BUILD.gn

Shared_pointer_cage has been supported on loong64 in
commit 5431c97a5706ce0d03b77d40dcb32537c55cd12e, but not turned on by
default. It's maybe time to add loong64 to the supporting list of
shared_pointer_cage in BUILD.gn

Bug: v8:11460
Change-Id: Iae17985f6b55a30bd030cadec31584e46e9bf2dd
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4444448
Commit-Queue: Liu Yu <liuyu@loongson.cn>
Reviewed-by: Igor Sheludko <ishell@chromium.org>
Cr-Commit-Position: refs/heads/main@{#87163}
---
 BUILD.gn | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index bca193c467..8a85594df5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -646,8 +646,8 @@ assert(
 assert(
     !v8_enable_pointer_compression_shared_cage || v8_current_cpu == "x64" ||
         v8_current_cpu == "arm64" || v8_current_cpu == "riscv64" ||
-        v8_current_cpu == "ppc64",
-    "Sharing a pointer compression cage is only supported on x64,arm64, ppc64 and riscv64")
+        v8_current_cpu == "ppc64" || v8_current_cpu == "loong64",
+    "Sharing a pointer compression cage is only supported on x64, arm64, ppc64, riscv64 and loong64")
 
 assert(!v8_enable_unconditional_write_barriers || !v8_disable_write_barriers,
        "Write barriers can't be both enabled and disabled")
-- 
2.35.1

